Makefile

```diff
@@ -9,25 +9,21 @@ else
     BUILD_DATE ?= $(shell date "$(DATE_FMT)")
 endif
 
-ifndef CGO_CPPFLAGS
-    export CGO_CPPFLAGS := $(CPPFLAGS)
-endif
-ifndef CGO_CFLAGS
-    export CGO_CFLAGS := $(CFLAGS)
-endif
-ifndef CGO_LDFLAGS
-    export CGO_LDFLAGS := $(LDFLAGS)
-endif
-
-GO_LDFLAGS := -X github.com/cli/cli/internal/build.Version=$(GH_VERSION) $(GO_LDFLAGS)
-GO_LDFLAGS := -X github.com/cli/cli/internal/build.Date=$(BUILD_DATE) $(GO_LDFLAGS)
-ifdef GH_OAUTH_CLIENT_SECRET
-	GO_LDFLAGS := -X github.com/cli/cli/internal/authflow.oauthClientID=$(GH_OAUTH_CLIENT_ID) $(GO_LDFLAGS)
-	GO_LDFLAGS := -X github.com/cli/cli/internal/authflow.oauthClientSecret=$(GH_OAUTH_CLIENT_SECRET) $(GO_LDFLAGS)
-endif
+CGO_CPPFLAGS ?= ${CPPFLAGS}
+export CGO_CPPFLAGS
+CGO_CFLAGS ?= ${CFLAGS}
+export CGO_CFLAGS
+CGO_LDFLAGS ?= ${LDFLAGS}
+export CGO_LDFLAGS
+
+GO_LDFLAGS += \
+	-X github.com/cli/cli/internal/build.Version=${GH_VERSION} \
+	-X github.com/cli/cli/internal/build.Date=${BUILD_DATE} \
+	${GH_OAUTH_CLIENT_ID:%=-X github.com/cli/cli/internal/authflow.oauthClientID=%} \
+	${GH_OAUTH_CLIENT_SECRET:%=-X github.com/cli/cli/internal/authflow.oauthClientSecret=%}
 
 bin/gh: $(BUILD_FILES)
-	@go build -trimpath -ldflags "$(GO_LDFLAGS)" -o "$@" ./cmd/gh
+	go build -trimpath -ldflags "${GO_LDFLAGS}" -o "$@" ./cmd/gh
```

Comment:
What is the difference between `$(GO_LDFLAGS)` and `${GO_LDFLAGS}` expressions?

Reply:
That is the format recommended by both GNU and *BSD, since a very long time ago... It's useful because then only GNU Make commands take the form `$(command ...)`.